spring-boot-project/spring-boot/src/main/java/org/springframework/boot/webservices/client/WebServiceTemplateBuilder.java

```diff
@@ -0,0 +1,1106 @@
+		candidates.put("org.springframework.http.client.ClientHttpRequestFactory",
+				ClientHttpRequestMessageSenderFactory.class);
+		MESSAGE_SENDER_FACTORY_CLASSES = Collections.unmodifiableMap(candidates);
+	}
+
+	private final Set<ClientInterceptor> interceptors;
+
+	private final Set<WebServiceTemplateCustomizer> internalCustomizers;
+
+	private final Set<WebServiceTemplateCustomizer> customizers;
+
+	private final Set<Supplier<? extends WebServiceMessageSender>> webServiceMessageSenderSuppliers;
+
+	private final Set<WebServiceMessageSenderCustomizer> webServiceMessageSenderCustomizers;
+
+	private final Marshaller marshaller;
+
+	private final Unmarshaller unmarshaller;
+
+	private final DestinationProvider destinationProvider;
+
+	private final Class<? extends TransformerFactory> transformerFactoryClass;
+
+	private final WebServiceMessageFactory messageFactory;
+
+	private final boolean detectWebServiceMessageSender;
+
+	public WebServiceTemplateBuilder(WebServiceTemplateCustomizer... customizers) {
+		this(Collections.emptySet(), Collections.emptySet(),
+				append(Collections.<WebServiceTemplateCustomizer>emptySet(), customizers),
+				Collections.emptySet(), Collections.emptySet(), null, null, null, null,
+				null, true);
+	}
+
+	private WebServiceTemplateBuilder(Set<ClientInterceptor> interceptors,
+			Set<WebServiceTemplateCustomizer> internalCustomizers,
+			Set<WebServiceTemplateCustomizer> customizers,
+			Set<Supplier<? extends WebServiceMessageSender>> webServiceMessageSenderSuppliers,
+			Set<WebServiceMessageSenderCustomizer> webServiceMessageSenderCustomizers,
+			Marshaller marshaller, Unmarshaller unmarshaller,
+			DestinationProvider destinationProvider,
+			Class<? extends TransformerFactory> transformerFactoryClass,
+			WebServiceMessageFactory messageFactory,
+			boolean detectWebServiceMessageSender) {
+		this.interceptors = interceptors;
+		this.internalCustomizers = internalCustomizers;
+		this.customizers = customizers;
+		this.webServiceMessageSenderSuppliers = webServiceMessageSenderSuppliers;
+		this.webServiceMessageSenderCustomizers = webServiceMessageSenderCustomizers;
+		this.marshaller = marshaller;
+		this.unmarshaller = unmarshaller;
+		this.destinationProvider = destinationProvider;
+		this.transformerFactoryClass = transformerFactoryClass;
+		this.messageFactory = messageFactory;
+		this.detectWebServiceMessageSender = detectWebServiceMessageSender;
+	}
+
+	/**
+	 * Set {@link ClientInterceptor ClientInterceptors} that should be used with the
+	 * {@link WebServiceTemplate}. Interceptors are applied in the order that they were
+	 * added after builder configuration has been applied.
+	 *
+	 * <b>Note!</b> Override existing interceptors
+	 * @param interceptors the interceptors to set
+	 * @return a new builder instance
+	 * @see WebServiceTemplate#setInterceptors(ClientInterceptor[])
+	 */
+	public WebServiceTemplateBuilder setInterceptors(ClientInterceptor... interceptors) {
```

Comment:
I prefer if we keep the same style as what we're doing with `RestTemplate` (i.e. `interceptors` and `additionalInterceptors`). I've already started a polish on my fork so need to act on it.